riklet/src/network/tap.rs

```diff
@@ -28,9 +30,57 @@ pub fn open_tap(config: &NetworkInterfaceConfig) -> Result<VirtioNet, NetworkInt
     .map_err(NetworkInterfaceError::CreateNetworkDevice)
 }
 
+/// FIXME: Find a better way to handle tap generation (using firecracker itself)
+/// This function creates a tap using legacy commands, it's not secure but it fixes a current issue with TAPs
+pub fn open_tap_shell(config: &NetworkInterfaceConfig) -> Result<String, NetworkInterfaceError> {
+    let mac_addr = generate_mac_addr();
+
+    if config.iface_name.len() > MAX_IFACE_NAME_LEN {
+        return Err(NetworkInterfaceError::InvalidInterfaceName);
+    }
+
+    let tap_output = Command::new("ip")
+        .args(["tuntap", "add", &config.iface_name, "mode", "tap"])
+        .output()
+        .map_err(|e| NetworkInterfaceError::ManageTap(e.to_string()))?;
+
+    if !tap_output.status.success() {
+        return Err(NetworkInterfaceError::ManageTap(format!(
+            "Tap creation failed, code {}, stderr: {}",
+            tap_output.status.code().unwrap(),
+            String::from_utf8(tap_output.stderr).unwrap()
+        )));
+    }
+
+    trace!("Shell tap create output: {:#?}", tap_output);
+    return Ok(config.iface_name.clone());
+}
+
+pub fn close_tap_shell(iface_name: &str) -> Result<(), NetworkInterfaceError> {
+    if iface_name.len() > MAX_IFACE_NAME_LEN {
+        return Err(NetworkInterfaceError::InvalidInterfaceName);
+    }
```

Comment:
Already handled on net.rs

Reply:
Handled only on `::new()` but `::new()` is not use for the moment

Reply:
Handled only on ::new() but ::new() is not used for the moment